### DISCUSSION TITLE
[SOLVED] How to download ADE key file from Windows 8.1?

### DISCUSSION THREAD

**Discussion by michaelbr51** [apprenticeharper/DeDRM_tools#1693] [General]
Sorry, after further reading, for Linux OS I need to add ADE key to the plugin, my ADE is in Windows 8.1 OS, tried to follow [this post](https://community.adobe.com/t5/digital-editions/key-license-damaged-adobe-digital-editions/m-p/7221287), but I'm afraid it didn't work (it exported as *.reg file and I can't find any key entry under adept), and when tried to import it in Calibre, it did not see the file. What I'm missing, how can I download/export the ADE key file in Windows 8.1 OS?
here's my environment
OS MX Linux 19.4
Calibre 5.16.1 (Linux)
DeDRM plugin 7.2.1 (Linux)
ADE 2.0 (Windows 8.1)

**Comment by michaelbr51** [comment]
I think I found part of the answer, but when I tried to run the adobekey.py, got this error:
`line 114, import winreg, no module named winreg`
my question: Is there any other script to generate adobe key for Windows 8.1 under VB? it seems the solution is for ADE installed under Wine, I'm using Windows in VirtualBox (couldn't install ADE under Wine, got lot of errors when trying to create prefix).

**Reply by ElleKayEm** [reply]
Try temporarily installing Windows calibre with DeDRM in VirtualBox.  If you can decrypt a book there, then you could save the key and import it into the Linux calibre/DeDRM setup.

**Reply by michaelbr51** [reply]
Thanks ElleKayEm for your reply, suppose that I could decrypt a book in VirtualBox, but how can I save the key? So far I've found only 1 post explaining how to generate a new key, couldn't find any post on how to save, do you mind share with me where I can find information how to save a key?
ps: any recommendation on which version of Calibre should I install in VirtualBox? (I'm using ADE 2.0 in Virtualbox).

**Reply by ElleKayEm** [reply]
In the Adobe section of DeDRM's customization, click the save button.

**Reply by michaelbr51** [reply]
Thanks so much ElleKayEm, it's working with ADE in VirtualBox, saved the key file as instructed. Thanks again.
